web/src/components/AnalyzerResult/Plugins.tsx

```diff
@@ -31,74 +34,87 @@ const Plugins = ({plugins, trace}: IProps) => {
   );
 
   return (
-    <Collapse expandIcon={({isActive = false}) => <CollapseIcon isCollapsed={isActive} />}>
-      {plugins.map(plugin => (
-        <S.PluginPanel
-          header={
-            <Space>
-              <LintScore width="35px" height="35px" score={plugin.score} />
-              <Typography.Text strong>{plugin.name}</Typography.Text>
-              <Typography.Text type="secondary">{plugin.description}</Typography.Text>
-            </Space>
-          }
-          key={plugin.name}
-        >
-          {plugin.rules.map(rule => (
-            <S.RuleContainer key={rule.name}>
-              <S.Column>
-                <S.RuleHeader>
-                  <Space>
-                    {rule.passed ? <S.PassedIcon $small /> : <S.FailedIcon $small />}
-                    <Tooltip title={rule.tips.join(' - ')}>
-                      <Typography.Text strong>{rule.name}</Typography.Text>
-                    </Tooltip>
-                  </Space>
-                </S.RuleHeader>
-                <Typography.Text type="secondary" style={{paddingLeft: 20}}>
-                  {rule.description}
-                </Typography.Text>
-              </S.Column>
+    <>
+      <S.SwitchContainer>
+        <Switch checked={onlyErrors} id="only_errors_enabled" onChange={() => setOnlyErrors(prev => !prev)} />
+        <label htmlFor="only_errors_enabled">Show only errors</label>
+      </S.SwitchContainer>
+
+      <S.StyledCollapse expandIcon={({isActive = false}) => <CollapseIcon isCollapsed={isActive} />}>
+        {plugins
+          .filter(plugin => !onlyErrors || plugin.score < MAX_SCORE)
```

Comment:
Should we consider having a selector or service function that returns only the data we want to render? Instead of having the business logic at the component level?

Reply:
Yep, good one. Done!